packages/admin-ui/src/components/welcome/features/EnableEthicalMetrics.tsx

```diff
@@ -12,43 +17,153 @@ export default function EnableEthicalMetrics({
   onBack?: () => void;
   onNext: () => void;
 }) {
+  const ethicalMetricsConfig = useApi.getEthicalMetricsConfig();
+  const [wasEthicalMetricsOn, setWasEthicalMetricsOn] = useState(false);
   const [ethicalMetricsOn, setEthicalMetricsOn] = useState(false);
   const [mail, setMail] = useState("");
   const [mailError, setMailError] = useState(false);
+  const [tgChannelId, setTgChannelId] = useState("");
+  const [tgChannelIdError, setTgChannelIdError] = useState(false);
+  const [validationMessage, setValidationMessage] = useState("");
+  const [tgAccordionOpen, setTgAccordionOpen] = useState(false);
 
-  // regex for email validation
+  // useEffect to populate email field when data is available
+  useEffect(() => {
+    const ethicalMetricsData = ethicalMetricsConfig.data;
+    if (ethicalMetricsData) {
+      ethicalMetricsData.mail && setMail(ethicalMetricsData.mail);
+      ethicalMetricsData.tgChannelId &&
+        setTgChannelId(ethicalMetricsData.tgChannelId);
+      setEthicalMetricsOn(ethicalMetricsData.enabled);
+      setWasEthicalMetricsOn(ethicalMetricsData.enabled);
+    }
+  }, [ethicalMetricsConfig.data]);
+
+  // effect and regex for email validation
   useEffect(() => {
     const regex = /\S+@\S+\.\S+/;
-    if (regex.test(mail)) {
+    if (regex.test(mail) || mail === "") {
       setMailError(false);
-      setEthicalMetricsOn(true);
     } else {
       setMailError(true);
-      setEthicalMetricsOn(false);
     }
   }, [mail]);
 
-  function onSetEnabeEthicalMetrics() {
-    if (ethicalMetricsOn)
-      api
-        .enableEthicalMetrics({ mail, tgChannelId: null, sync: false })
-        .catch(e => {
-          console.error(`Error on autoUpdateSettingsEdit : ${e.stack}`);
-        });
-    onNext();
+  // effect and regex for telegram channelId validation
+  useEffect(() => {
+    const regex = /^-100\d{10}$/;
+    if (regex.test(tgChannelId) || tgChannelId === "") {
+      setTgChannelIdError(false);
+    } else {
+      setTgChannelIdError(true);
+    }
+  }, [tgChannelId]);
+
+  async function enableEthicalMetricsSync({
+    mailValue,
+    tgChannelIdValue
+  }: {
+    mailValue?: string | null;
+    tgChannelIdValue?: string | null;
+  }) {
+    try {
+      setValidationMessage("Enabling ethical metrics...");
+      await api.enableEthicalMetrics({
+        mail: mailValue || mail,
+        tgChannelId: tgChannelIdValue || tgChannelId,
+        sync: true
```

Comment:
Have you test the behaviour of setting this with option `sync` to true? Take into account that this might be the onboarding and might take some time the user would not like to wait during setting up dappnode